### PR TITLE
fix: request prevented when re-downgrade subscribe

### LINKED ISSUE #4782: downgrade twice will not execute
downgrade -> resume -> downgrade will not execute success

image

its a prevent-replay problem

### PR DESCRIPTION
**Pull request by darkskygit**
fix #4782

### CODE REVIEW

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/plans/plan-card.tsx [hunk 1]
```diff
@@ -532,19 +534,20 @@ const ResumeAction = ({
...
-  // allow replay request on network error until component unmount
-  const idempotencyKey = useMemo(() => nanoid(), []);
-
   const resume = useCallback(() => {
     trigger(
       { idempotencyKey },
       {
         onSuccess: data => {
+          // refresh idempotency key
+          setIdempotencyKey(nanoid());
```
Comment (joooye34) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/plans/plan-card.tsx:550]: how about error occur?

Comment (darkskygit) [reply packages/frontend/core/src/components/affine/setting-modal/general-setting/plans/plan-card.tsx:550]: In our scenario, there are four possible errors:

1. network error
2. server error
3. stripe gateway exception
4. stripe believes that there is something wrong with the information transmitted

These four errors occur in order. When the first three errors occur, the ideapotency key does not take effect, so the client can rest assured to use the same key request.
When the last error occurs, it means that the subscription information of the corresponding user is problematic (for example, he has not started subscribing, or the subscription has ended), but tries to call the resume subscription.
At this time, it is appropriate for the stripe to directly use the cache response corresponding to the ideapotencyKey.

**Review by joooye34 — APPROVED** [review]

**Review by joooye34 — APPROVED** [review]

**Comment by joooye34** [comment]
Commit need to be signed, seems need recommit -S

**Comment by darkskygit** [comment]
> Commit need to be signed, seems need recommit -S

fixed

**Pull request merged**